filesystem/flysystem/FlysystemAssetStore.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+namespace SilverStripe\Filesystem\Flysystem;
+
+use Config;
+use Injector;
+use InvalidArgumentException;
+use League\Flysystem\Exception;
+use League\Flysystem\Filesystem;
+use SilverStripe\Filesystem\Storage\AssetNameGenerator;
+use SilverStripe\Filesystem\Storage\AssetStore;
+
+/**
+ * Asset store based on flysystem Filesystem as a backend
+ */
+class FlysystemAssetStore implements AssetStore {
+
+	/**
+	 * @var Filesystem
+	 */
+	protected $filesystem;
```

Comment:
I'm for consistency! (Ironic, I know).

But if there are no consistent getters/setters then we can't use private because ancestors are unable to access key properties. however, if the properties are accessible via methods, this is fine.